app/src/main/assets/chart.html

```diff
@@ -77,35 +77,25 @@
             {% set class = summaryValue | format_values(row.item.cssClass) %}
             {% set style = summaryValue | format_values(row.item.cssStyle) %}
             <td id="cell-{{id}}-{{column.start.millis}}"
-                class="{{column.start == nowColumnStart ? 'now' : ''}} {{class}}"
-                style="{{style}}" onclick="popup({{row.item.label | js}},
-              {% if points is empty %}
-                []
-              {% else %}
-                [ {%- for point in points -%}
-                  [{{point.value | format_value(row.item.captionFormat) | js}},
-                   {{point.time | format_time('HH:mm') | js}}],
-                {%- endfor -%} ]
-              {% endif %} )">
-              {% if points is not empty %}
-                {% if row.item.type == 'text_icon' %}
-                  <div onclick="alert(''
-                        {%- for point in points -%}
-                          + '---- '
-                          + {{point.time | format_time('dd MMM \'at\' HH:mm') | js}}
-                          + ' ----\n'
-                          + {{point.value.text | js}}
-                          + '\n\n'
-                        {%- endfor -%}); event.stopPropagation();">&#x1f4dd;</div>
-                {% else %}
-                  {% set output = summaryValue | format_value(row.item.format) %}
-                  {% if output is empty %}
-                    {% set output = summaryValue | format_value(row.item.captionFormat) %}
-                  {% endif %}
-                  {{output}}
-                {% endif %}
-              {% endif %}
-            </td>
+              class="{{column.start == nowColumnStart ? 'now' : ''}} {{class}}"
+              style="{{style}}"
+              onclick="{% if points is not empty%}
+                       {% if (row.item.type).string != 'text_icon' %}
```

Comment:
There's a lot of duplicated logic in these conditionals. Could you clean that up with a variable?